rclpy/src/rclpy/service.cpp

```diff
@@ -182,25 +135,41 @@ service_take_request(py::capsule pyservice, py::object pyrequest_type)
     throw std::bad_alloc();
   }
 
+  py::tuple result_tuple(2);
   rcl_ret_t ret = rcl_take_request_with_info(
-    &(srv->service), header.get(), taken_request.get());
+    rcl_service_.get(), header.get(), taken_request.get());
   if (ret == RCL_RET_SERVICE_TAKE_FAILED) {
-    return py::none();
-  } else if (ret != RCL_RET_OK) {
+    result_tuple[0] = py::none();
+    result_tuple[1] = py::none();
+    return result_tuple;
+  } else if (RCL_RET_OK != ret) {
     throw RCLError("service failed to take request");
   }
 
   py::list result_list(2);
   result_list[1] = py::capsule(header.release(), "rmw_service_info_t");
 
-  PyObject * pytaken_request_c = rclpy_convert_to_py(taken_request.get(), pyrequest_type.ptr());
-  if (!pytaken_request_c) {
-    throw py::error_already_set();
-  }
-  result_list[0] = py::reinterpret_steal<py::object>(pytaken_request_c);
+  result_list[0] = convert_to_py(taken_request.get(), pyrequest_type);
   // result_list now owns the message
   taken_request.release();
 
-  return std::move(result_list);
+  return result_list;
```

Comment:
Looks like there's a new clang warning here.

https://ci.ros2.org/job/ci_osx/11897/clang-tidy/new/source.dce8798b-3637-471a-8bb9-b4b3454f5026/#156

I think it can be fixed by returning `result_tuple` instead of `result_list` (to match the `py::tuple` return type)